tests/build_spec_generator/cli_command_parser/test_gradle_cli_command.py

```diff
@@ -0,0 +1,156 @@
+            "gradlew clean build -Pskip.signing -Pgnupg.skip ",
+            id="test_properties_equal_checking",
+        ),
+        pytest.param(
+            "gradlew clean build -Dorg.gradle.caching=true -PmyProperty=boo",
+            "gradlew clean build -Dorg.gradle.caching=true -PmyProperty=boo",
+            id="test_properties_with_values_equal_checking",
+        ),
+        pytest.param(
+            "gradlew clean build -x test -x boo",
+            "gradlew clean build -x test -x boo",
+            id="test_excluded_tasks",
+        ),
+    ],
+)
+def test_comparing_gradle_cli_command_equal(
+    gradle_cli_parser: GradleCLICommandParser,
+    this: str,
+    that: str,
+) -> None:
+    """Test comparing two equal GradleCLICommand objects."""
+    this_command = gradle_cli_parser.parse(this.split())
+    that_command = gradle_cli_parser.parse(that.split())
+    assert this_command == that_command
+
+
+@pytest.mark.parametrize(
+    ("this", "that"),
+    [
+        ("gradle clean build", "gradle clean"),
+        ("gradle", "gradlew"),
+        ("gradle clean build", "gradle clean build -PmyProperty=true"),
+        ("gradle clean build -Dorg.gradle.caching=true", "gradle clean build -Dorg.gradle.caching=false"),
+        ("gradle clean build -Dorg.gradle.caching=true", "gradle clean build -Dorg.gradle.caching"),
+        ("gradle clean build", "gradle clean build -c settings.gradle"),
+        ("gradle build", "gradle build -x test"),
+        # We persist the order which the task names are put into the excluded list.
+        # Therefore the order of the -x options is important.
+        ("gradle build -x test -x boo", "gradle build -x boo -x test"),
+        ("gradle build --no-build-cache", "gradle build --build-cache"),
+    ],
+)
+def test_comparing_gradle_cli_command_unequal(
+    gradle_cli_parser: GradleCLICommandParser,
+    this: str,
+    that: str,
+) -> None:
+    """Test comparing two unequal GradleCLICommand objects."""
+    this_command = gradle_cli_parser.parse(this.split())
+    that_command = gradle_cli_parser.parse(that.split())
+    assert not this_command == that_command
+
+
+@pytest.mark.parametrize(
+    ("command", "that"),
+    [
+        (
+            "gradle clean build -x test --debug --stacktrace -Dorg.gradle.caching=true",
+            True,
+        ),
+        (
+            "gradle clean build -x test --debug --stacktrace -Dorg.gradle.caching=true",
+            ["boo", "foo"],
+        ),
+        (
+            "gradle clean build -x test --debug --stacktrace -Dorg.gradle.caching=true",
+            {"boo", "foo"},
+        ),
+    ],
+)
+def test_comparing_gradle_cli_command_unequal_types(
+    gradle_cli_parser: GradleCLICommandParser,
+    command: str,
+    that: Any,
+) -> None:
+    """Test comparing MavenCLICommand with another incompatible type oject."""
+    this_command = gradle_cli_parser.parse(command.split())
+    assert not this_command == that
```

Comment:
What is this testing? It seems to be comparing a GradleCLICommand object against a bool, list, and set.